impl/src/test/groovy/io/jsonwebtoken/impl/DefaultJwtParserTest.groovy

```diff
@@ -69,7 +70,8 @@ class DefaultJwtParserTest {
             }
         }
         def p = new DefaultJwtParser().deserializeJsonWith(deserializer)
-        assertSame deserializer, p.deserializer
+        assertTrue("Expected wrapping deserializer to be instance of JwtDeserializer", p.deserializer instanceof JwtDeserializer )
```

Comment:
@Marcono1234 you have a lot of good catches in this review - thanks so much for chiming in!

As for asserting deprecated methods, I see two sides to this:  1) `JwtParser#deserializeJsonWith` is deprecated, so no new functionality/patches are guaranteed for such methods, so new behavior doesn't have to be asserted, versus 2) we implemented it, so we should check it just for sanity in our own unit tests.

I'd vote in favor of approach 2 just because I like to be exhaustively thorough.  @bdemers any thoughts?  Can we add a quick test?

Reply:
My comment was slightly misleading. The modified test, on which this review comment is added, is already testing the behavior for the deprecated `JwtParser#deserializeJsonWith` (by calling `new DefaultJwtParser().deserializeJsonWith(deserializer)`, see code). So that is already covered by these changes.

The point was that it would be good to perform the same assertion on the non-deprecated `DefaultJwtParserBuilder` as well.